alf/environments/suite_carla.py

```diff
@@ -753,14 +753,27 @@ def get_current_time_step(self, current_frame):
         # -------- Infraction 2: running red light --------
         red_light_id = self._alf_world.is_running_red_light(self._actor)
         if red_light_id is not None and red_light_id != self._prev_violated_red_light_id:
-            logging.info("actor=%d frame=%d RED_LIGHT" % (self._actor.id,
-                                                          current_frame))
+            speed = self._get_agent_speed()
+            logging.info("actor=%d frame=%d RED_LIGHT speed %2.1f" %
+                         (self._actor.id, current_frame, speed))
             reward_vector[Player.REWARD_RED_LIGHT] = 1.
             info['red_light'] = np.float32(1.0)
-
-            reward -= min(
-                self._max_red_light_penalty,
-                Player.PENALTY_RATE_RED_LIGHT * max(0., self._episode_reward))
+            if self._terminate_upon_infraction != "redlight":
+                reward -= min(
+                    self._max_red_light_penalty,
+                    Player.PENALTY_RATE_RED_LIGHT * max(
+                        0., self._episode_reward))
+
+            elif self._terminate_upon_infraction != "redlight":
```

Comment:
If am reading this correctly, the `elif` and `if` conditions are the same?

Reply:
My bad. This was a bug introduced when preparing for this PR.